R/axe-generics.R

```diff
@@ -1,38 +1,47 @@
 #' Butcher an object.
 #'
 #' Reduce the size of a model object so that it takes up less memory on disk.
-#' Currently the model object is stripped down to the point that only the
-#' minimal components necessary for the `predict` function to work remain.
+#' Currently, the model object is stripped down to the point that only the
+#' minimal components necessary for the \code{predict} function to work remain.
 #' Future adjustments to this function will be needed to avoid removal of
 #' model fit components to ensure it works with other downstream functions.
 #'
-#' @param x model object
-#' @param ... any additional arguments related to axing
+#' @param x Model object.
+#' @param verbose Print information each time an axe method is executed.
+#'  Notes how much memory is released and what functions are
+#'  disabled. Default is \code{TRUE}.
+#' @param ... Any additional arguments related to axing.
 #'
-#' @return axed model object with new butcher subclass assignment
+#' @return Axed model object with new butcher subclass assignment.
 #' @export
-butcher <- function(x, ...) {
+butcher <- function(x, verbose = TRUE, ...) {
   old <- x
-  pre <- options(usethis.quiet = TRUE)
-  x <- axe_call(x, ...)
-  x <- axe_ctrl(x, ...)
-  x <- axe_data(x, ...)
-  x <- axe_env(x, ...)
-  x <- axe_fitted(x, ...)
-  pre <- options(usethis.quiet = FALSE)
-  # TODO: figure out inheritance dynamically..
-  assess_object(old, x, working = c("predict"))
+
+  temp_file <- tempfile(fileext = ".txt")
+  sink(file = temp_file)
+  x <- axe_call(x, verbose = verbose, ...)
+  x <- axe_ctrl(x, verbose = verbose, ...)
+  x <- axe_data(x, verbose = verbose, ...)
+  x <- axe_env(x, verbose = verbose, ...)
+  x <- axe_fitted(x, verbose = verbose, ...)
+  sink()
```

Comment:
I feel like we might be able to come up with a better way around this. What if `axe_call()` always returned `x` with an attribute `butcher_disabled` which was a vector of character strings of the disabled function names?

Then inside `butcher()` you would set `verbose = FALSE` for all of the individual axe functions, but in that final `if (verbose)` check below, you could just extract out the `attr(x, "butcher_disabled")` object and pass that along to `assess_object()`?

Reply:
Oh, I like that idea. Thus these functions are carried along with the butchered model object  

Will update to reflect!